my-app/src/main/java/com/mycompany/app/App.java

```diff
@@ -22,4 +43,23 @@ public IInteger BigIncrement(IInteger rascal_value) {
         // Return value that can be used in Rascal
         return vf.integer(java_value);
     }
+
+    public IString frink_parse(IString text)
+    {
+        String unsafe_user_input = text.getValue();
+        String result = "";
+
+        try
+        {
+           result = interp.parseString(unsafe_user_input);
+        }
+        catch (FrinkEvaluationException e)
+        {
+            // Rethrow as Rascal exception
+            throw RuntimeExceptionFactory.illegalArgument(text, "Input cannot be converted by Frink");
+        }
```

Comment:
This is how a Java exception is caught, and a new Rascal exception is created for it.